pkg/apis/core/validation/validation.go

```diff
@@ -4495,6 +4495,11 @@ func ValidateServiceUpdate(service, oldService *core.Service) field.ErrorList {
 // ValidateServiceStatusUpdate tests if required fields in the Service are set when updating status.
 func ValidateServiceStatusUpdate(service, oldService *core.Service) field.ErrorList {
 	allErrs := ValidateObjectMetaUpdate(&service.ObjectMeta, &oldService.ObjectMeta, field.NewPath("metadata"))
+	if service.Spec.Type != core.ServiceTypeLoadBalancer &&
+		len(service.Status.LoadBalancer.Ingress) > 0 {
```

Comment:
I know of some use-cases where it's useful to set `service.Status.LoadBalancer.Ingress` without the type being `LoadBalancer`. It's maybe far fetched, but `status.loadbalancer.ingress.ip` is useful for binding Services to arbitrary VIPs where you can't use ClusterIP since it must be in the service range and you can't use ExternalIPs cause internal traffic is dropped

Reply:
I'd also consider this a breaking change since I'm sure there's a non-trivial amount of users setting service status without the type being load balancer.

Reply:
On one hand, normal users should NOT have access to service Status.  What you are describing sounds...awful.

On the other hand, I have been trying to be even-handed about saying "no, that's a breaking change", even when I like the change.  So I think Andrew's right - we can't actually do this.  We can log and even throw an event when we observe this, but we can't break it.  There's not even a protocol for doing this over some extended period of time.  The rule is simply "we don't break the API" (on purpose, at least).

:(

Reply:
> "we don't break the API" (on purpose, at least).

yeah, that was I was wondering too, this seems to be like this since forever :shrug: 

> ExternalIPs cause internal traffic is dropped

That is another thing I stumbled on recently, https://github.com/kubernetes/kubernetes/issues/95785#issuecomment-722928906
is it on purpose or an implementation detail? 
I can't see a reason to not allow internal traffic to ExternalIPs